repository/confluent-schema-registry/docs/v1.0/security.md

```diff
@@ -0,0 +1,103 @@
+
+### Configuration for KUDO Kafka for TLS Authentication
+
+Follow the steps of [Enabling KUDO Kafka for TLS encryption](https://github.com/kudobuilder/operators/blob/master/repository/kafka/docs/latest/security.md#enabling--tls-encryption) and enable the TLS authentication by setting the parameter `SSL_AUTHENTICATION_ENABLED` to `true` as:
+
+```
+$ kubectl kudo install kafka \
+  --instance=kafka \
+    -p AUTHORIZATION_ENABLED=true \
+    -p AUTHORIZATION_ALLOW_EVERYONE_IF_NO_ACL_FOUND=true \
+    -p TRANSPORT_ENCRYPTION_ENABLED=true \
+    -p TRANSPORT_ENCRYPTION_ALLOW_PLAINTEXT=false \
+    -p SSL_AUTHENTICATION_ENABLED=true
+```
+
+Now, we can connect Confluent Schema Registry with Kafka Brokers using TLS Authentication, similar to the [TLS encryption](#enabling-tls-encryption):
+
+```
+$ kubectl kudo install confluent-schema-registry \
+    --instance=schema-registry  \
+    -p KAFKA_BOOTSTRAP_SERVERS="SSL://kafka-kafka-0.kafka-svc.default.svc.cluster.local:9095,SSL://kafka-kafka-1.kafka-svc.default.svc.cluster.local:9095,SSL://kafka-kafka-2.kafka-svc.default.svc.cluster.local:9095" \
+    -p TRANSPORT_ENCRYPTION_ENABLED=true \
+    -p TLS_SECRET_NAME=confluent-schema-registry-tls
+```
+
+## Authorization
+
+The KUDO Kafka service supports Kafka’s ACL-based authorization system.  To use Kafka’s ACLs with Confluent Schema Registry as a client, TLS authentication must be enabled as detailed [above](#authentication).
+
+### Enable Authorization with TLS
+
+#### Prerequisites
+
+* Completion of TLS authentication as detailed [above](#authentication).
+
+Install the KUDO Kafka service with the following options in addition to your own (remember, TLS authentication must be enabled):
+
+```
+$ kubectl kudo install kafka \
+  --instance=kafka \
+    -p ZOOKEEPER_URI=zk-zookeeper-0.zk-hs:2181,zk-zookeeper-1.zk-hs:2181,zk-zookeeper-2.zk-hs:2181 \
+    -p BROKER_CPUS=200m \
+    -p BROKER_COUNT=3 \
+    -p BROKER_MEM=800m \
+    -p DISK_SIZE=10Gi \
+    -p AUTHORIZATION_ENABLED=true \
+    -p AUTHORIZATION_ALLOW_EVERYONE_IF_NO_ACL_FOUND=false \
+    -p AUTHORIZATION_SUPER_USERS="User:SchemaRegistryClient" \
+    -p TRANSPORT_ENCRYPTION_ENABLED=true \
+    -p TRANSPORT_ENCRYPTION_ALLOW_PLAINTEXT=false \
+    -p SSL_AUTHENTICATION_ENABLED=true
+```
+
+NOTE: Here, we need to supply `AUTHORIZATION_SUPER_USERS` parameter as the client name that we provide to our Confluent Schema Registry client. It can be configured by `SSL_AUTHENTICATION_CLIENT_NAME` parameter of Schema Registry and its default value is `SchemaRegistryClient`.
+
+Now, we can connect Confluent Schema Registry with Kafka Brokers using TLS Authentication, similar to the [TLS encryption](#enabling-tls-encryption), by supplying the client name to the `SSL_AUTHENTICATION_CLIENT_NAME` parameter, as configured in Kafka Broker's ACL during KUDO Kafka's deployment:
```

Comment:
@vibhujain  The instructions in this paragraph is kinda repeated with the paragraph above

Reply:
I know but we need to supply Client name on both the sides as parameters. I need to refactor the docs for more clarity and to avoid repetition but reinstantiate that we need it as a parameter while deploying both services.

Reply:
it can be better this way
```
Now, we can connect Confluent Schema Registry with Kafka Brokers using TLS Authentication, similar to the [TLS encryption](#enabling-tls-encryption):
```

Reply:
While keeping  your suggestion made for L92, I propose L94 can be:
`Suggestive example of connecting KUDO Schema Registry with KUDO Kafka using TLS authentication with client name specified as above:`

Reply:
It will look like:
 
NOTE: To grant super user permission to KUDO Confluent Schema Registry, the client name that is provided to Schema Registry needs to be added to the `AUTHORIZATION_SUPER_USERS` parameter for KUDO Kafka. The client name can be configured using `SSL_AUTHENTICATION_CLIENT_NAME` parameter which has a default value  `SchemaRegistryClient`. 

Suggestive example of connecting KUDO Schema Registry with KUDO Kafka using TLS authentication with client name specified as above:

```
$ kubectl kudo install confluent-schema-registry \
    --instance=schema-registry  \
    -p KAFKA_BOOTSTRAP_SERVERS="SSL://kafka-kafka-0.kafka-svc.default.svc.cluster.local:9095,SSL://kafka-kafka-1.kafka-svc.default.svc.cluster.local:9095,SSL://kafka-kafka-2.kafka-svc.default.svc.cluster.local:9095" \
    -p TRANSPORT_ENCRYPTION_ENABLED=true \
    -p TLS_SECRET_NAME=confluent-schema-registry-tls \
    -p SSL_AUTHENTICATION_CLIENT_NAME=SchemaRegistryClient
```

Reply:
~~Suggestive~~ Example of connecting KUDO Schema Registry with KUDO Kafka using TLS authentication with client name specified as above: